client/lib/wpcom-undocumented/lib/me.js

```diff
@@ -98,10 +98,11 @@ UndocumentedMe.prototype.setPeerReferralLinkEnable = function ( enable, callback
 	return this.wpcom.req.post( args, callback );
 };
 
-UndocumentedMe.prototype.backupCodes = function ( callback ) {
+UndocumentedMe.prototype.backupCodes = function ( password, callback ) {
 	const args = {
 		apiVersion: '1.1',
 		path: '/me/two-step/backup-codes/new',
+		body: { password },
```

Comment:
Any security concerns with the fact that the user password is going to that endpoint in plain text? 

In general, is it concerning that we're adding another place where we're expecting the user password, which could potentially be another attack vector? Or is that just fine, considering that the endpoint would require the user to be authenticated in the first place?

Reply:
What sort of attack vectors do you think might be possible if we use this endpoint?

An alternative would be to send this to `wp-login`, but not sure if this would be really different to what we are doing here. To get a valid result on this endpoint, the request needs to be authenticated (with a "fresh" 2fa cookie). We'll only add a password validation step along with some throttling to limit how many times one could test the password.

Reply:
I didn't mean anything specific, and I don't have any specific concerns. I just wanted to get your point of view as someone who has a better understanding of security than me 😉

Reply:
Thanks for the question :) I had a somewhat similar concern (passing around the user password in different parts of code) when I was making the change, and I was curious if you have something particular in mind.

Reply:
FWIW [we are doing the same](https://github.com/Automattic/wp-calypso/blob/migrate/undocumented-publicize/client/state/login/actions/login-user.js#L34) when logging in the user already, so I think we should be good.

Reply:
GitHub web UI has something similar: before entering some sensitive parts of the site, you need to reenter your password, and it leads to a `post( '/sessions/sudo', { sudo_password } )` request.

The only concern is that there's now one more REST endpoint that handles passwords, and one more thing to audit and be careful about.

But as long the request is a `POST` and is encrypted, it's technically OK.

Reply:
Do you maybe know what other endpoints handle passwords? The only one I know is the settings endpoint to update the current user password. I agree however that we might probably consider having a similar feature to harden some areas of the application.

On an unrelated note, do you maybe know why the e2e tests might be failing. This change doesn't touch any dependency.

Reply:
> Do you maybe know what other endpoints handle passwords? The only one I know is the settings endpoint to update the current user password. I agree however that we might probably consider having a similar feature to harden some areas of the application.

We directly use the canonical backend WP.com login as an endpoint when logging in - see https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/state/login/utils.js#L165.

> On an unrelated note, do you maybe know why the e2e tests might be failing. This change doesn't touch any dependency.

It seems the docker build failed for this branch, but a rebase of the branch should fix it, it doesn't seem related.